src/main.rs

```diff
@@ -28,14 +30,21 @@ use dirs::home_dir;
 use log::{error, trace, warn};
 use lsp_server::{Connection, ErrorCode, Message, Notification, Request, RequestId, Response};
 use lsp_types::{
-    notification::{Notification as _, *},
-    request::{Request as RequestTrait, *},
-    OneOf,
-    *,
+    notification::{DidChangeTextDocument, DidOpenTextDocument, Notification as _},
+    request::{
+        Completion, DocumentLinkRequest, Formatting, GotoDefinition, Rename,
+        Request as RequestTrait, SelectionRangeRequest,
+    },
+    CompletionItem, CompletionOptions, CompletionTextEdit, Diagnostic, DiagnosticSeverity,
+    DidChangeTextDocumentParams, DidOpenTextDocumentParams, DocumentLink, DocumentLinkOptions,
+    DocumentLinkParams, Location, OneOf, Position, PublishDiagnosticsParams, Range, RenameParams,
+    SelectionRangeProviderCapability, ServerCapabilities, TextDocumentPositionParams,
+    TextDocumentSyncCapability, TextDocumentSyncKind, TextDocumentSyncOptions, TextEdit, Url,
+    WorkDoneProgressOptions, WorkspaceEdit,
```

Comment:
I don't think that this improves the readability. It seems as if that's a great way to get avoidable merge-conflicts especially if multiple folks edit this. So while adding `clippy` here by default is a great thing, I'd prefer to exclude this rule :)

Reply:
I was considering this as well, but thought I'd just go along with clippy for the first round and see if there were any objections :slightly_smiling_face:

Reply:
I'm personally in favor of using an explicit (for example) `lsp_types::CompletionItem` syntax over `use lsp_types::*` when there are many imports. But I don't want to nitpick too much, so I'm in favor of either just accepting this entire PR (then later refactoring slightly) or reverting this small chunk of the diff, whichever @Ma27 prefers